aws/resource_aws_quicksight_group_test.go

```diff
@@ -0,0 +1,241 @@
+	resourceName := "aws_quicksight_group.default"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckQuickSightGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSQuickSightGroupConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckQuickSightGroupExists(resourceName, &group),
+					testAccCheckQuickSightGroupDisappears(&group),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckQuickSightGroupExists(resourceName string, group *quicksight.Group) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		awsAccountID, namespace, groupName, err := resourceAwsQuickSightGroupParseID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).quicksightconn
+
+		input := &quicksight.DescribeGroupInput{
+			AwsAccountId: aws.String(awsAccountID),
+			Namespace:    aws.String(namespace),
+			GroupName:    aws.String(groupName),
+		}
+
+		output, err := conn.DescribeGroup(input)
+
+		if err != nil {
+			return err
+		}
+
+		if output.Group == nil {
+			return fmt.Errorf("Quick Sight Group (%s) not found", rs.Primary.ID)
+		}
+
+		*group = *output.Group
+
+		return nil
+	}
+}
+
+func testAccCheckQuickSightGroupDestroy(s *terraform.State) error {
+	meta := testAccProvider.Meta()
+	client := meta.(*AWSClient).quicksightconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_quicksight_group" {
+			continue
+		}
+
+		awsAccountID, namespace, groupName, err := resourceAwsQuickSightGroupParseID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = client.DescribeGroup(&quicksight.DescribeGroupInput{
+			AwsAccountId: aws.String(awsAccountID),
+			Namespace:    aws.String(namespace),
+			GroupName:    aws.String(groupName),
+		})
+		if !isAWSErr(err, quicksight.ErrCodeResourceNotFoundException, "") {
+			return fmt.Errorf("Quick Sight Group '%s' was not deleted properly", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckQuickSightGroupDisappears(v *quicksight.Group) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).quicksightconn
+
+		arn, err := arn.Parse(aws.StringValue(v.Arn))
+		if err != nil {
+			return err
+		}
+
+		parts := strings.SplitN(arn.Resource, "/", 3)
+
+		input := &quicksight.DeleteGroupInput{
+			AwsAccountId: aws.String(arn.AccountID),
+			Namespace:    aws.String(parts[1]),
+			GroupName:    v.GroupName,
+		}
+
+		if _, err := conn.DeleteGroup(input); err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccAWSQuickSightGroupConfig(rName string) string {
+	return fmt.Sprintf(`
+provider "aws" {
```

Comment:
For now, please omit hardcoded `provider` configurations in test configurations and rely on the environment variable handling. We'll eventually be circling around to fix the problems caused by this type of testing setup. 😄 See also: #8316 and #8983

Reply:
removed.